subsys/nfc/rpc/common/nfc_rpc_common.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "nfc_rpc_common.h"
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+
+void nfc_rpc_decode_error(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+			  void *handler_data)
+{
+	nrf_rpc_rsp_decode_i32(group, ctx, handler_data);
+}
+
+void nfc_rpc_report_decoding_error(uint8_t cmd_evt_id)
+{
+	nrf_rpc_err(-EBADMSG, NRF_RPC_ERR_SRC_RECV, &nfc_group, cmd_evt_id,
+		    NRF_RPC_PACKET_TYPE_CMD);
+}
+
+void nfc_rpc_decode_parameter(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+			      void *handler_data)
+{
+	struct nfc_rpc_param_getter *getter = (struct nfc_rpc_param_getter *)handler_data;
+	const void *data;
+
+	data = nrf_rpc_decode_buffer_ptr_and_size(ctx, getter->length);
```

Comment:
Verify this doesn't exceed the maximum length?

Reply:
@Damian-Nordic, sorry, forgot to add my comment.
According to NFC API requested memory cannot be larger than the maximum size that was requested.
https://github.com/nrfconnect/sdk-nrfxlib/blob/main/nfc/include/nfc_t2t_lib.h#L156-L170 
I really do not think NFC RPC should manage a situation that shouldn't happen.
If it happens, this is the bug and should be fixed in code rather than handled by API serializer.

Reply:
ok, I thought it was a good idea to write a decoder in a defensive way but I may be overpedantic :)